modules/apps/headless/headless-builder/headless-builder-impl/src/main/java/com/liferay/headless/builder/internal/model/listener/HeadlessBuilderObjectEntryModelListener.java

```diff
@@ -59,84 +60,141 @@ public void onAfterUpdate(
 		_schedulePublication(objectEntry);
 	}
 
-	private long _getAPIApplicationId(ObjectEntry objectEntry) {
-		if (_isObjectDefinition("L_API_APPLICATION", objectEntry)) {
+	@Override
+	public void onBeforeRemove(ObjectEntry objectEntry)
+		throws ModelListenerException {
+
+		try {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					objectEntry.getObjectDefinitionId());
+
+			if (StringUtil.equals(
```

Comment:
What if someone removes just and endpoint?

Reply:
Yeah, we should take care of it. Thanks @ccorreagg!!